### PR TITLE
Add documentation for authentication token expirations

### PR DESCRIPTION
**Pull request by JarrettSpiker**
### What
Adds documentation for authentication token expiration, including changes to the API to set/display token expiration.

### Why
An upcoming feature will allow user, team, and organization tokens to be created with an expiration time set. We want to encourage users to set an expiration on new tokens.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] (N/A) Redirects have been added for moved, renamed, or deleted pages. This requires a separate PR in the [`terraform-website` repository](https://github.com/hashicorp/terraform-website) `redirects.next.js` file.
- [ ] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] (N/A) Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] (N/A) New pages have metadata (page name and description) at the top.
- [x] (N/A) New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] (N/A) UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx [hunk 1]
```diff
@@ -87,7 +87,7 @@ The following chart illustrates the various access levels for the supported API
 | Modify organizations               |      🔶     |             |                     |
 | Manage organization tokens         |      🔶     |             |                     |
 | View audit trails                  |             |             |          🔵         |
-| Invite users to organiation        |      🔶     |      🔶     |          🔵         |
+| Invite users to organization       |      🔶     |      🔶     |          🔵         |
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx:90]: Great catch! 🕵️‍♂️

File: website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx [hunk 2]
```diff
@@ -98,3 +98,11 @@ The following chart illustrates the various access levels for the supported API
 | Manage run tasks                   |      🔶     |      🔶     |          🔵          |
 | **Modules**                        |             |             |                     |
 | Manage Terraform modules           |      🔶     | 🔵 (owners) |                     |
+
+## Token Expiration
+
+User, team, and organization tokens may be created with an expiration date and time set. Once this time has passed, the token will no longer be treated as valid, and may not be used to authenticate to any API.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx:104]: Non-blocking suggestion: I think we can omit the "set" here. Additionally, what if we explicitly state that API requests will fail if made with an expired token (even though it's implied) -- thoughts?:
```suggestion
User, team, and organization tokens may be created with an expiration date and time. Once this time has passed, the token will no longer be treated as valid, and may not be used to authenticate to any API. Any API requests made with an expired token will result in failure.
```

File: website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx [hunk 3]
```diff
@@ -98,3 +98,11 @@ The following chart illustrates the various access levels for the supported API
...
+
+HashiCorp recommends setting an expiration on all new authentication tokens. Creating tokens which will expire once they are no longer needed helps reduce the risk that valid tokens may be leaked, or that token owners will forget to delete tokens meant for delegated use once their intended purpose is complete.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx:106]: Non-blocking suggestion: What if we rephrase this slightly?:
```suggestion
HashiCorp recommends setting an expiration on all new authentication tokens. Creating tokens with an expiration date helps reduce the risk that valid tokens may be leaked, or that token owners will forget to delete tokens meant for delegated use once their intended purpose is complete.
```

File: website/docs/cloud-docs/users-teams-organizations/users.mdx [hunk 4]
```diff
@@ -121,11 +121,14 @@ API tokens are required for the following tasks:
...
 Protect your tokens carefully because they contain the same permissions as your user account. For example, if you belong to a team with permission to read and write variables for a workspace, another user could use your API token to authenticate as your user account and also edit variables in that workspace. Refer to [permissions](/terraform/cloud-docs/users-teams-organizations/permissions) for more details.
 
+Terraform Cloud also recommends protecting your tokens by giving them an expiration date and time on token creation. Refer to [API Token Expiration](/terraform/cloud-docs/users-teams-organizations/api-tokens#token-expiration) for details.
+
 #### Creating a Token
 
  To create a new token:
   1. Click **Create an API token**. The **Create API token** box appears.
   1. Enter a **Description** that explains what the token is for and click **Create API token**.
+  1. Optionally, enter an expiration date or time for the token, or choose to create a token which will never expire. Token expiration will always be displayed in the UI relative to your current time zone.
```
Comment (juliannatetreault) [inline website/docs/cloud-docs/users-teams-organizations/users.mdx:131]: Non-blocking suggestion: Thoughts about rephrasing this slightly?:
```suggestion
  1. Optionally, enter an expiration date or time for the token, or choose to create a token which will never expire. In the UI, token expirations will always be displayed in your current time zone.
```

**Review by juliannatetreault — APPROVED** [review]
Thorough! I left a few non-blocking suggestions, but otherwise think that this reads super nicely🌟 

One nitpicky suggestion that I had that I couldn't highlight in the code was updating the org token description to be consistent with how we refer to the team and user token endpoints in the token descriptions (`/authentication-token` and `/authentication-tokens`) by including the backslash. We'd have to do that for the [org tokens description](https://github.com/hashicorp/terraform-docs-common/blob/6b7f1b93d89f1e54ed8b8af94faac3b07c87c9cf/website/docs/cloud-docs/api-docs/organization-tokens.mdx#organization-token-api) (line 4 in `organization-tokens.mdx` in the codebase) if we decided to move forward with the update!

File: website/docs/cloud-docs/api-docs/team-tokens.mdx [hunk 5]
```diff
@@ -46,17 +46,43 @@ Generates a new team token and overrides existing token if one exists.
 | :----- | :----------------------------------- |
 | POST   | /teams/:team_id/authentication-token |
 
+This endpoint returns the secret text of the created authentication token. A token value is only shown upon creation, and cannot be recovered later.
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/team-tokens.mdx:49]:
```suggestion
This endpoint returns the secret text of the new authentication token. A token value is only returned upon creation, and cannot be recovered later.
```

File: website/docs/cloud-docs/api-docs/user-tokens.mdx [hunk 6]
```diff
@@ -198,10 +201,11 @@ This POST endpoint requires a JSON object with the following properties as a req
...
-| ----------------------------- | ------ | ------- | ----------------------------------- |
-| `data.type`                   | string |         | Must be `"authentication-tokens"`.  |
-| `data.attributes.description` | string |         | The description for the User Token. |
+| Key path                      | Type   | Default | Description                                                                                                     |
+| ----------------------------- | ------ | ------- | --------------------------------------------------------------------------------------------------------------- |
+| `data.type`                   | string |         | Must be `"authentication-tokens"`.                                                                              |
+| `data.attributes.description` | string |         | The description for the User Token.                                                                             |
+| `data.attributes.expired-at`  | string | `null`  | The UTC date and time that the User Token will expire. If omitted or set to `null` the token will never expire. |
```
Comment (robin-norwood) [inline website/docs/cloud-docs/api-docs/user-tokens.mdx:208]:
```suggestion
| `data.attributes.expired-at`  | string | `null`  | The UTC date and time that the User Token will expire, in ISO 8601 format. If omitted or set to `null` the token will never expire. |
```

**Pull request merged**